### DISCUSSION TITLE
Trouble with load balancing socket.io with redis and nginx

### DISCUSSION THREAD

**Discussion by lenguyenkhoi21** [socketio/socket.io#4178] [Q&A] [answered]
I've 3 server socket.io, using redis adapter and nginx for load balancing. When I test in my computer, there is only server is using, and the last two server socket.io can't be routed to ? How to test with three server routing ? Here is my code: https://github.com/lenguyenkhoi21/socketio-scale

**Comment by darrachequesne** [comment, marked as answer]
It seems the session stickiness is based on the IP, so you need clients from different IP adresses.

Please check the example here: https://github.com/socketio/socket.io/tree/master/examples/cluster-nginx

**Reply by lenguyenkhoi21** [reply]
Yes, I'm using same at you.

**Reply by darrachequesne** [reply]
@lenguyenkhoi21 in that case, if you run `docker-compose up -d --scale=client=10 client` to create multiple client containers (each with a different IP address), and then check the logs with `docker-compose logs client`, you should see that each client is connected to a different server:

```
client_3         | connected to Paul
client_3         | connected to Ringo
client_3         | connected to Ringo
client_8         | connected to Ringo
client_6         | connected to George
client_6         | connected to Ringo
client_6         | connected to Ringo
client_4         | connected to Paul
client_4         | connected to Ringo
client_2         | connected to Paul
client_2         | connected to Ringo
client_7         | connected to Ringo
client_10        | connected to George
client_10        | connected to Ringo
client_5         | connected to John
client_5         | connected to Ringo
client_9         | connected to John
client_9         | connected to Ringo
client_1         | connected to Ringo
```

I think that's the easiest way to test the load balancing in local.

**Reply by lenguyenkhoi21** [reply]
@darrachequesne  I have my solution. Just remove nginx from docker-compose, export the port of 3 server instance from docker, and running nginx in my window. I'm using more devices and it load-balancing
